### ISSUE TITLE
[Feature Request] Model loading progress display or callback?

### ISSUE THREAD

**Issue by juntaosun** [microsoft/onnxruntime#17796]
### Describe the feature request

Does onnxruntime.InferenceSession need to add a progress indicator for loading the model? With the development of technology, there are more and more large models, and they often need to wait for a long time when loading. If you add a progress callback or indicator, this It will make the user experience of onnxruntime better.

-----------

onnxruntime.InferenceSession 是否需要添加一个加载模型的进度指示，随着技术的发展，现在有越来越多的大模型，它们在加载时往往需要等待很长的时间，如果添加一个进度回调或指示，这将让onnxruntime的用户体验变的更好。

### Describe scenario use case

In this way, when developing applications and switching models, we can show the user the loading progress of the model, but the interface will still be unresponsive for a few seconds. Although the loading of small models is completed quickly, large models with a G volume will have to wait. With the iteration of technology, future models may be larger. Just like the game loading local resources, a real-time progress value is fed back to facilitate the improvement of application development and operating experience. Thank you~

-----------

这样在开发应用时，在切换模型时，我们可以向用户展示模型的加载进度，而还是几秒的界面无响应。虽然小模型加载很快就完成，但是上G体积的大模型就会等待，随着技术的迭代，以后的模型也许体积更大。就像游戏加载本地资源一样，反馈一个实时进度值，以方便改善应用的开发与操作体验。谢谢~

**Comment by pranavsharma** [comment]
Typically, ORT sessions are created (models are loaded) as part of the application/process startup, not on individual user requests. This way the model loading time is hidden from users. This is even more true for large models. Is there a reason you're loading large models on individual user queries? You can always load the model in a separate thread and have a condition variable notify the invoking thread when it's done.

**Comment by mjsb212** [comment]
I agree, this would be a useful feature -- When loading ORT sessions either at startup, and/or switching models after startup, it would be good to report the progress because on some devices this takes a while with large models.
